api/services/EmailTemplates/forgottenUsername.ejs

```diff
@@ -0,0 +1,3 @@
+<p>
+  Your username for your account at <a href="http://<%= siteUrl %>"><%= siteUrl %></a> is <%= recipientName %>.
```

Comment:
Because if we didn't, that would also allow anyone to figure out whether a given email address is registered, by trying to register another account with that email address.

Yeah, sending an email with all the usernames is probably best.